### PR TITLE
Updating Pytest criteria

### PR DESCRIPTION
**Pull request by sfc-gh-tmathew**

### CODE REVIEW

**Comment by sfc-gh-tmathew** [comment]
@sfc-gh-twhite Added dev and master branch pytest github actions. We have dependency reviews triggered for each PR. Will pull this to lower branches once merged.

File: .github/workflows/dev-pytest.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (sfc-gh-twhite) [inline .github/workflows/dev-pytest.yml:16]: For dev, let's pick one version of Python for now. This should speed up tests where only simple changes are occurring. 

```
    steps:
      - uses: actions/checkout@v4
      - name: Set up Python 3.11
        uses: actions/setup-python@v4
        with:
          python-version: "3.11"
```

**Review by sfc-gh-twhite — CHANGES_REQUESTED** [review]
I think one change, but this looks great!

**Comment by sfc-gh-twhite** [comment]
And should we add a workflow_dispatch so we can trigger these ad-hoc?

**Comment by sfc-gh-tmathew** [comment]
> And should we add a workflow_dispatch so we can trigger these ad-hoc?

Added workflow dispatch for manual workflow against Dev

**Comment by sfc-gh-tmathew** [comment]
@sfc-gh-twhite Added the requested changes.

**Review by sfc-gh-twhite — APPROVED** [review]

**Pull request merged**